src/Mbstring/Mbstring.php

```diff
@@ -871,6 +871,49 @@ public static function mb_str_pad(string $string, int $length, string $pad_strin
         }
     }
 
+    public static function mb_ucfirst(string $string, ?string $encoding = null): string {
+        if (null === $encoding) {
+            $encoding = self::mb_internal_encoding();
+        }
+
+        try {
+            $validEncoding = @self::mb_check_encoding('', $encoding);
+        } catch (\ValueError $e) {
+            throw new \ValueError(sprintf('mb_ucfirst(): Argument #2 ($encoding) must be a valid encoding, "%s" given', $encoding));
+        }
```

Comment:
`self::mb_internal_encoding()` performs validation checks when setting internal values, so adding a `try-catch` to the `else` avoids a small amount of overhead. What do you think?

```suggestion
        } else {
            try {
                $validEncoding = @self::mb_check_encoding('', $encoding);
            } catch (\ValueError $e) {
                throw new \ValueError(sprintf('mb_ucfirst(): Argument #2 ($encoding) must be a valid encoding, "%s" given', $encoding));
            }
        }
```